opm/input/eclipse/Schedule/Action/State.hpp

```diff
@@ -20,74 +20,142 @@
+        ///
+        /// \param[in] that Object against which \code *this \endcode will
+        /// be tested for equality.
+        ///
+        /// \return Whether or not \code *this \endcode is the same as \p that.
+        bool operator==(const MatchSet& that) const;
 
-    static RunState serializationTestObject()
-    {
-        RunState rs;
-        rs.run_count = 100;
-        rs.last_run = 123456;
-        return rs;
-    }
+        /// Convert between byte array and object representation.
+        ///
+        /// \tparam Serializer Byte array conversion protocol.
+        ///
+        /// \param[in,out] serializer Byte array conversion object.
+        template <class Serializer>
+        void serializeOp(Serializer& serializer)
+        {
+            serializer(this->wells_);
+        }
 
-    bool operator==(const RunState& other) const
-    {
-        return this->run_count == other.run_count &&
-               this->last_run == other.last_run;
-    }
+        friend class State;
 
-    template<class Serializer>
-    void serializeOp(Serializer& serializer)
-    {
-        serializer(this->run_count);
-        serializer(this->last_run);
-    }
+    private:
+        /// Set of triggering wells.
+        std::vector<std::string> wells_{};
+    };
 
-    std::size_t run_count{};
-    std::time_t last_run{};
-};
+    /// Record ActionX Run
+    ///
+    /// \param[in] action Action object.
+    ///
+    /// \param[in] sim_time Time at which action object ran.
+    ///
+    /// \param[in] result Result of evaluating the action triggers,
+    /// including any matching entities such as wells.
+    void add_run(const ActionX& action, std::time_t sim_time, const Result& result);
 
-public:
-    void add_run(const ActionX& action, std::time_t sim_time, Result result);
+    /// Record PyAction Run
+    ///
+    /// \param[in] action PyAction object.
+    ///
+    /// \param[in] result Result of evaluating the PyAction.
     void add_run(const PyAction& action, bool result);
+
+    /// Retrieve number of times an action has run
+    ///
+    /// \param[in] action Action object.
+    ///
+    /// \return Number of times \p action has run.
     std::size_t run_count(const ActionX& action) const;
+
+    /// Retrieve timestamp of the last time an action run.
+    ///
+    /// Will throw an exception of type std::invalid_argument if the action
+    /// has never run, i.e., if \code run_count(action) == 0 \endcode.
+    ///
+    /// \param[in] action Action object.
+    ///
+    /// \return Time point of \p action's last execution.
     std::time_t run_time(const ActionX& action) const;
-    std::optional<Result> result(const std::string& action) const;
+
+    /// Retrieve set of matching entities from the last time an action ran.
+    ///
+    /// \param[in] action Action name.
+    ///
+    /// \return Set of matching entities.  Nullptr if no such set
+    /// exists--e.g., if the action did not yet run or if there were not
```

Comment:
no

Reply:
> no

Absolutely right.  I've fixed this in the latest update.